core/src/main/scala/org/apache/spark/input/PortableDataStream.scala

```diff
@@ -47,7 +47,7 @@ private[spark] abstract class StreamFileInputFormat[T]
   def setMinPartitions(sc: SparkContext, context: JobContext, minPartitions: Int) {
     val defaultMaxSplitBytes = sc.getConf.get(config.FILES_MAX_PARTITION_BYTES)
     val openCostInBytes = sc.getConf.get(config.FILES_OPEN_COST_IN_BYTES)
-    val defaultParallelism = sc.defaultParallelism
+    val defaultParallelism = Math.max(sc.defaultParallelism, minPartitions)
```

Comment:
We should have a test case; otherwise, we could hit the same issue again.

Reply:
BTW, it is easy to add such a test case. We can even test the behaviors of the boundary cases. cc @srowen @HyukjinKwon @MaxGekk @jiangxb1987

Reply:
I think it's hard to test, technically, because `setMinPartitions` is only a hint. In the case of `binaryFiles` we know it will put a hard limit on the number of partitions, but it isn't true of other implementations. We can still make a simple test for all of these, it just may be asserting behavior that could change in the future in Hadoop, though I strongly doubt it would.

Reply:
I agree it is hard to test. I appreciate If anyone can give me some hints of how to do these (how to verify and where to put my test cases).

Reply:
Would you mind following up with a test that just asserts that asking for, say, 20 partitions results in 20 partitions? This is technically too specific as a test, but is probably fine for now.

Reply:
From the codes, you can see the calculation is just the intermediate result and this method won't return any value. Checking the split size does not make sense for this test case because it depends on multiple variables and this is just one of them.

Reply:
```
      sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local")
        .set(config.FILES_OPEN_COST_IN_BYTES.key, "0")
        .set("spark.default.parallelism", "1"))

      println(sc.binaryFiles(dirpath1, minPartitions = 50).getNumPartitions)
      println(sc.binaryFiles(dirpath1, minPartitions = 1).getNumPartitions)
```

It is not hard to verify whether the parameter `minPartitions` takes an effect. Currently, the description of this parameter is not clear. We need to document it clear which factors impact the actual number of partitions; otherwise, users will not understand how to use it.